packages/kmod-6.1-nvidia/kmod-6.1-nvidia.spec

```diff
@@ -5,6 +5,17 @@
 %global spdx_id %(bottlerocket-license-tool -l %{_builddir}/Licenses.toml spdx-id nvidia)
 %global license_file %(bottlerocket-license-tool -l %{_builddir}/Licenses.toml path nvidia -p ./licenses)
 
+# With the split of the firmware binary from firmware/gsp.bin to firmware/gsp_ad10x.bin
+# and firmware/gsp_tu10x.bin the file format changed from executable to relocatable.
+# The __spec_install_post macro will by default try to strip all binary files.
+# Unfortunately the strip used is not compatible with the new file format.
+# The macro seems to have an obscure past and nobody really seems to know how it
+# came to be and what its overall purpose is. The process of disabling stripping
+# seems to be to redefine __spec_install_post to nil. However, as we rely on our
+# own __arch_install_post macro that gets invoced through __spec_install_post to
+# generate the _cross_attribution_file. So redefine to directly invoce __arch_install_post.
+%define __spec_install_post %{__arch_install_post}
```

Comment:
This way of skipping stripping is a bit heavy-handed/unspecific as it cuts out whatever else was defined in `__spec_install_post`. Right now that's:

```
[root@f78bf2a149ef builder]# rpm --eval '%{__os_install_post}'

     
    /usr/lib/rpm/brp-compress 
    
    /usr/lib/rpm/brp-strip /usr/bin/x86_64-bottlerocket-linux-gnu-strip 
    /usr/lib/rpm/brp-strip-comment-note /usr/bin/x86_64-bottlerocket-linux-gnu-strip /usr/bin/x86_64-bottlerocket-linux-gnu-objdump 
     
    /usr/lib/rpm/redhat/brp-strip-lto /usr/bin/x86_64-bottlerocket-linux-gnu-strip 
    /usr/lib/rpm/brp-strip-static-archive /usr/bin/x86_64-bottlerocket-linux-gnu-strip 
    /usr/lib/rpm/check-rpaths 
    /bin/true 
    /usr/lib/rpm/brp-remove-la-files 
    
    /usr/lib/rpm/redhat/brp-python-bytecompile "" "1" "0" 
     
    /usr/lib/rpm/redhat/brp-python-hardlink 
 


```

If only stripping is problematic, you should be able to make that a no-op via `%define __strip /usr/bin/true`. You could even replace that with a wrapper that only skips stripping the firmware files.

Reply:
It is indeed a drastic measure. Admittedly, I went for the easiest way here. I will play around with the suggestions and see what the overall difference is in the different cases.

Reply:
So I have built this now once with just disabling the stripping and once with the patch as is. As far as I can tell from diffing the resulting rpms with diffoscope the only differences are what is expected from building the same rpm at two different times:

* timestamp differences
* build time differences
* hash differences because the buildtimes are included in the calculation of those checksums

Either way, I will go with the proposed less intrusive change.